### PR TITLE
refactor(location)!: throw error if state is unknown

### LINKED ISSUE #1736: location.zipCodeByState should throw for unknown states
### Clear and concise description of the problem

Currently, if the given state is not known, then a random zip code is returned.
Usually, if I pass a value to a method, then I expect it to use it and tell me if the value is wrong.

### Suggested solution

Throw if there is no data for that state.

### Alternative

Keep the method as is.

### Additional context

https://next.fakerjs.dev/api/location.html#zipcodebystate

### PR DESCRIPTION
**Pull request by 0916dhkim**
Fixes #1736 

- #1736

### CODE REVIEW

File: src/modules/location/index.ts [hunk 1]
```diff
@@ -64,9 +64,9 @@ export class LocationModule {
     const zipRange = this.faker.definitions.location.postcode_by_state?.[state];
     if (zipRange) {
       return String(this.faker.number.int(zipRange));
+    } else {
+      throw new Error(`${state} is invalid.`);
```
Comment (ST-DDT) [inline src/modules/location/index.ts]: Please throw a FakerError instead.
Maybe also add a list with allowed keys and a hint where these keys are comming from.
E.g. `faker.definitions.location.postcode_by_state only contains data for [...]`

**Review by ST-DDT — COMMENTED** [review]
If you have trouble running the tests locally, you can use `CI=true pnpm run test watch`.

**Review by matthewmayer — CHANGES_REQUESTED** [review]
This documentation needs to be updated to reflect that an error would now be thrown. Also I think it could be make clearer this doesn't work in the default `en` locale, but only in `en_US`.

File: src/modules/location/index.ts (hunk 1 above)
Comment (matthewmayer) [reply src/modules/location/index.ts]: you might want to have different errors for the 
1. where the locale doesn't support postcode_by_state at all (e.g. en, de)
1. where the locale does support postcode_by_state, but this is an unknown state (say XX in en_US).

Comment (ST-DDT) [reply src/modules/location/index.ts]: We will handle the first part centrally: https://github.com/faker-js/faker/issues/893#issuecomment-1397329605

- #893

Comment (matthewmayer) [reply src/modules/location/index.ts]: hmm i think this is a weird edge case though where the fallback locale (en) doesn't contain the data, which is unusual.

Comment (0916dhkim) [reply src/modules/location/index.ts]: @ST-DDT Thanks for the review 🙏 
I'll use `FakeError` instead + attach available keys.

Comment (0916dhkim) [reply src/modules/location/index.ts]: I have fixed the code to throw FakerError and updated the docstring, but I'm not sure how to make the tests pass. It looks like `en` locale is used as default in `seededTests`, and `zipCodeByState` will always throw with `en`.

**Comment by Shinigami92** [comment]
The unresolved conflict brings an issue: the `state` parameter is required and therefore we have two options:

1. We just allow `zipCodeByState({ state: '...', .../* potential other options */ })`
2. We allow `zipCodeByState({ state: '...', .../* potential other options */ })` as well as `zipCodeByState('state', { /* potential other options */ })`

Once we said we only want to have method calls (outside the helpers module) that can always be called without passing any arguments. In this case state is always required otherwise it would always throw an error.

We will discuss this in an upcoming meeting.

**Comment by ST-DDT** [comment]
**Team Decision**

- We want to merge zipCode and zipCodeByState into a single method instead (https://github.com/faker-js/faker/issues/1346#issuecomment-1407728227)
- It will still throw an error if the state doesn't have a config.

---

Thanks for your contribution anyway!

**Pull request closed without merging**